core/scripts/pptr-run-devtools.js

```diff
@@ -183,6 +183,11 @@ async function waitForLighthouseReady() {
       }
     }
   }
+
+  // Ensure the emulation model is ready before Lighthouse starts by enabling device emulation.
+  // `waitForLighthouseReady` can be run multiple times, so this should be the final action.
```

Comment:
I don't have enough background to follow the reasoning here. Can you explain more?

Reply:
`waitForLighthouseReady` is run in a loop until it returns successfully without throwing an error. If an error occurred after this line, we would toggle the device emulation multiple times. Toggling device emulation is not idempotent so this could be a problem.

Maybe theres a better function we could use that's not a toggle.